File: new-algorithm/src/javaalgo/implementation/boj2525.java
package javaalgo.implementation;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class boj2525 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int A = Integer.parseInt(st.nextToken());
        int B = Integer.parseInt(st.nextToken());
        int C = Integer.parseInt(br.readLine());

        if (B + C >= 60) {
            A += (B + C) / 60;
            B = (B + C) % 60;

            if (A >= 24) A %= 24;
        } else {
            B += C;
        }

        System.out.println(A + " " + B);
    }
}

File: new-algorithm/src/kotlinalgo/implementation/boj2525.kt
package kotlinalgo.implementation

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val st = StringTokenizer(readLine())

    var A = Integer.parseInt(st.nextToken())
    var B = Integer.parseInt(st.nextToken())
    var C = Integer.parseInt(readLine())

    when {
        B + C >= 60 -> {
            A += (B + C) / 60
            B = (B + C) % 60
            when {
                A >= 24 -> A %= 24
                else -> A
            }
        }
        else -> B += C
    }

    println("$A $B")
}